### ISSUE TITLE
📈 Bản tin xu hướng AI mã nguồn mở 2026-08-19

### ISSUE THREAD

**Issue by thanhtantran** [thanhtantran/agents-radar#345]
# Xu hướng AI Mã nguồn mở 2026-08-19

> Nguồn: GitHub Trending + GitHub Search API | Thời gian tạo: 2026-08-19 02:00 UTC

---

# Báo cáo Xu hướng AI Mã nguồn mở - 19/08/2026

## 📊 Tóm tắt hôm nay

Ngày 19/08/2026 chứng kiến sự bùng nổ của **AI Agent Infrastructure** với hơn 15 dự án liên quan đến agent harness, memory systems, và multi-agent frameworks. Đặc biệt nổi bật là xu hướng **local-first AI** với các giải pháp chạy hoàn toàn offline trên hardware cá nhân. Cộng đồng đang chuyển từ các giải pháp cloud sang **self-hosted, privacy-first** systems.

**Con số ấn tượng**: 
- MoneyPrinterTurbo dẫn đầu với +2,304 ⭐ trong ngày
- 8/12 repo trending liên quan trực tiếp đến AI agents
- Sự xuất hiện mạnh mẽ của Rockchip NPU ecosystem cho embedded AI

---

### Trending hôm nay

**🔥 harry0703/MoneyPrinterTurbo** (+2,304 ⭐)
- Python-based automated workflow tạo video HD từ keywords
- Tích hợp AI models để tự động hóa content creation
- Use case: Content creators, marketing automation

**chaitanyagiri/munder-difflin** (+306 ⭐)
- Local multi-agent harness
- TypeScript implementation
- Focus: Orchestration cho nhiều agents làm việc cùng lúc

### Từ GitHub Search (7 ngày)

**🌟 NousResearch/hermes-agent** (232K ⭐)
- "The agent that grows with you" - self-evolving agent
- Khả năng học và phát triển theo thời gian

**shareAI-lab/learn-claude-code** (74K ⭐)
- "Bash is all you need" - minimal agent harness
- Educational: Xây dựng từ 0 đến 1
- Tương tự Claude Code architecture

**zhayujie/CowAgent** (46K ⭐)
- Open-source super AI assistant
- Task planning, tool execution, self-evolution
- Multi-model, multi-channel support
- Formerly chatgpt-on-wechat - đang mở rộng scope

**HKUDS/nanobot** (47K ⭐)
- Ultra-lightweight framework
- WebUI + tools + memory + MCP
- Multi-agent workflows + automation
- Self-hosted personal agent

**siyuan-note/siyuan** (45K ⭐)
- Knowledge workspace cho humans + AI agents
- Privacy-first, self-hosted
- Collaboration between human and agents

---

### Agent Development Tools

**akitaonrails/ai-memory** (+648 ⭐)
- Rust-based long-term memory solution
- Cross-vendor agent handoff
- Giải quyết vấn đề: Agent CLI memory persistence

**volcengine/OpenViking** (+213 ⭐)
- Self-evolving Context Database
- Unify: Agent Memory + Knowledge RAG + Skills
- All-in-one solution cho agent infrastructure

**affaan-m/ECC** (240K ⭐)
- Agent harness performance optimization
- Skills, instincts, memory, security
- Research-first development
- Hỗ trợ: Claude Code, Codex, Opencode, Cursor

**thedotmack/claude-mem** (91K ⭐)
- Persistent context across sessions
- AI-powered compression
- Works với mọi major agent platforms
- Giải quyết: Session continuity problem

### LLM Infrastructure

**ollama/ollama** (178K ⭐)
- Hỗ trợ models mới: Kimi-K2.6, GLM-5.2, MiniMax, DeepSeek, gpt-oss
- Local LLM inference
- Growing model ecosystem

**langchain-ai/langchain** (144K ⭐)
- "The agent engineering platform"
- Rebranding từ LLM framework → Agent platform
- Signal: Industry shift toward agents

**open-webui/open-webui** (149K ⭐)
- User-friendly interface cho Ollama + OpenAI API
- Self-hosted solution
- 149K stars = massive adoption

**jundot/omlx** (+370 ⭐)
- LLM inference cho Apple Silicon
- Continuous batching + SSD caching
- macOS menu bar management
- Niche: Mac developers

### Development Platforms

**langgenius/dify** (152K ⭐)
- Agentic workflows + RAG pipelines
- Cloud, VPC, or self-hosted
- Team collaboration focus
- Prototype → Production without rebuilding

**CherryHQ/cherry-studio** (50K ⭐)
- Smart chat + autonomous agents + 300+ assistants
- Unified access to frontier LLMs
- Productivity studio approach

**FlowiseAI/Flowise** (55K ⭐)
- Visual AI Agent builder
- No-code/low-code approach
- 55K stars = strong visual builder demand

### Specialized Tools

**firecrawl/firecrawl** (169K ⭐)
- Context API to interact with web at scale
- Search, scrape, interact
- Essential cho web-enabled agents

**browser-use/browser-use** (109K ⭐)
- Make websites accessible for AI agents
- Automate online tasks
- Web automation layer

**mukul975/Anthropic-Cybersecurity-Skills** (+730 ⭐)
- 817 structured cybersecurity skills
- 6 frameworks mapped: MITRE ATT&CK, NIST CSF 2.0, etc.
- Works với 20+ platforms
- Apache 2.0 license

---

## 🧠 Models & Training

**huggingface/transformers** (164K ⭐)
- Framework cho SOTA models
- Text, vision, audio, multimodal
- Inference + training

**Significant-Gravitas/AutoGPT** (186K ⭐)
- Vision: Accessible AI for everyone
- Tools để focus on what matters
- 186K stars = iconic project

---

### Content Creation

**hugohe3/ppt-master** (47K ⭐)
- AI turns documents → PowerPoint decks
- Native shapes, transitions, animations
- Data-backed charts
- Audio narration from speaker notes
- Template support

**OpenCut-app/OpenCut** (+192 ⭐)
- Open-source CapCut alternative
- Video editing with AI

### Productivity

**santifer/career-ops** (65K ⭐)
- Open-source AI job search
- Scan portals, evaluate with A-F rubric (1.0-5.0 score)
- Tailor CV, track applications
- Runs locally in AI coding CLI
- Use case: Job hunting automation

**ZhuLinsen/daily_stock_analysis** (63K ⭐)
- LLM-driven multi-market stock analysis
- Multi-source data + real-time news
- Decision dashboard + automated notifications
- Zero-cost scheduled runs
- Chinese + English support

### Development Tools

**agalwood/Motrix** (+609 ⭐)
- Full-featured download manager
- TypeScript implementation

**basecamp/omarchy** (+356 ⭐)
- Beautiful, Modern & Opinionated Linux
- Shell-based distribution
- Focus: User experience

**public-apis/public-apis** (+1,005 ⭐)
- Collective list of free APIs
- Essential resource cho developers

---

## 🔍 RAG & Knowledge

**Shubhamsaboo/awesome-llm-apps** (133K ⭐)
- 100+ AI Agents + Agent Skills + RAG Apps
- Free and open source
- Comprehensive collection

**infiniflow/ragflow** (88K ⭐)
- Leading open-source RAG engine
- Fusion: RAG + Agent capabilities
- Superior context layer cho LLMs

**datawhalechina/hello-agents** (73K ⭐)
- Tiếng Trung: "从零开始构建智能体"
- Educational từ zero to hero
- Agent principles + practice

**headroomlabs-ai/headroom** (66K ⭐)
- Compress: tool outputs, logs, files, RAG chunks
- 20% fewer tokens cho coding agents
- 60-95% fewer tokens cho JSON
- Library + proxy + MCP server
- Critical: Token cost optimization

**Mintplex-Labs/anything-llm** (64K ⭐)
- "Stop renting your intelligence. Own it."
- Local-first agent experience
- Self-hosted solution

**mem0ai/mem0** (63K ⭐)
- Universal memory layer cho AI Agents
- Agent memory standard

**run-llama/llama_index** (51K ⭐)
- Leading document agent + OCR platform
- Rebranding: Framework → Platform

**jeecgboot/JeecgBoot** (47K ⭐)
- Enterprise AI low-code platform
- Tiếng Trung: Generate code với natural language
- AI Skills: 画流程, 设计表单, 生成报表, 大屏
- Built-in: AI chat, knowledge base, workflow, MCP plugins
- 90% reduction in repetitive work

---

### Rockchip NPU Ecosystem

**jaylfc/taOS** (491 ⭐) 🔥
- Self-hosted AI agent OS
- Runs on Orange Pi, Raspberry Pi, Mac mini, gaming PC
- Offline by default, cloud by choice
- Full web desktop + app store
- Auto-clustering across consumer hardware
- **Breakthrough**: Unified OS cho heterogeneous hardware

**jaylfc/taosmd** (77 ⭐)
- Local-first AI memory
- Runs offline on 8GB+ RAM machines
- Zero-loss verbatim archive + knowledge graph
- Framework-agnostic
- Companion cho taOS

**GatekeeperZA/RKLLM-API-Server** (19 ⭐)
- OpenAI-compatible API cho Rockchip NPU
- RK3588/RK3576 support
- Orange Pi + Open WebUI integration
- Critical: API standardization

**Hanzo-Huang/rkllm-docker** (9 ⭐)
- Dockerized RKLLM runtime
- OpenAI-compatible API
- Deployment simplification

**isac322/rkmon** (7 ⭐)
- Real-time hardware monitor TUI
- Like htop cho GPU, NPU, VPU, RGA, thermal zones
- RK3588 SBCs (Rock 5B+)
- Developer tool

**YeWenxuan64/Edge_Inferencer** (2 ⭐)
- Unified edge AI inference engine
- One Python API: Rockchip NPU + Qualcomm HTP + ONNX
- Auto-detects .rknn / .bin / .onnx
- Plug-and-play

**YeWenxuan64/Edge_ModelDeploy** (0 ⭐, mới)
- CV model deployment toolchain
- PyTorch/TensorFlow → ONNX → Edge NPUs
- Rockchip RKNPU + Qualcomm HTP
- INT8/INT4 + mixed-precision quantization

**oRKLLM/ork-driver** (1 ⭐)
- Clean-room userspace matmul library
- Rockchip NPU native implementation
- Low-level optimization

**clfang666/Panther-x2-NPU-VPU** (4 ⭐)
- Ubuntu 24.04 Armbian cho Panther X2
- Rockchip BSP 6.1
- VPU/NPU acceleration

**gahingwoo/mesa-rk3576** (0 ⭐, mới)
- Mesa/Teflon rocket driver
- RK3576 NPU support
- Mesa 25.3.0 base

**zouri/rtop** (0 ⭐, mới)
- Read-only terminal monitor cho Rockchip Linux
- CPU, memory, thermal, GPU, NPU, DDR, power, fan
- Comprehensive monitoring

### Orange Pi Projects

**MichaIng/DietPi** (6,197 ⭐)
- Lightweight OS cho SBCs
- "Lightweight justice for your single-board computer"
- Mature project

**RaspAP/raspap-webgui** (5,213 ⭐)
- Full-featured wireless router setup
- Debian-based devices
- Orange Pi compatible

**quintus-lab/openwrt-rockchip** (242 ⭐)
- ImmortalWrt 25.12 builder
- NanoPi R2S/R4S + Orange Pi R1 Plus

**lmambr2/moneypenny** (4 ⭐)
- Self-hosted AI + music assistant
- TeamSpeak 6 server
- Two editions: SBC (Orange Pi/RK3588) + Server (x86+GPU)
- Whisper STT + Piper British TTS + local Gemma
- No cloud required

**dustbowlair424/NavCore-Pixhawk** (6 ⭐)
- GPS-denied INS navigation cho drones
- Pixhawk Cube Orange + Raspberry Pi 4
- MAVLink protocol

**NawfalMotii79/PLFM_RADAR** (+192 ⭐)
- Open-source, low-cost 10.5 GHz PLFM phased array RADAR
- PLSQL implementation
- Hardware project

---

### 1. **Agent Harness Wars đã bắt đầu**
- Từ "agent harness" xuất hiện trong 7+ repos trending
- Key players: ECC, claude-mem, learn-claude-code, munder-difflin
- Competition: Performance optimization vs memory persistence vs multi-agent orchestration
- **Insight**: Cộng đồng đang standardize agent infrastructure layer

### 2. **Local-First Renaissance**
- taOS, taosmd, ollama, anything-llm, open-webui
- Slogan: "Stop renting your intelligence. Own it."
- Privacy-first, self-hosted, offline-capable
- **Driver**: Data privacy concerns + vendor lock-in fears + cost optimization

### 3. **Rockchip NPU Explosion** 🚀
- 10 repos mới trong 7 ngày về RKNPU/RKLLM
- Ecosystem đang hình thành: API servers, monitoring tools, Docker images, drivers
- Hardware: Orange Pi, Rock 5B+, RK3588, RK3576
- **Breakthrough**: OpenAI-compatible APIs cho edge NPUs
- **Implication**: Democratization of AI inference

### 4. **Token Cost Optimization**
- headroom: 20-95% token reduction
- claude-mem: AI-powered context compression
- **Context**: GPT-4, Claude pricing pressure
- **Trend**: Compression layers becoming standard middleware

### 5. **Agent Memory as Critical Infrastructure**
- ai-memory, claude-mem, mem0, OpenViking
- Problem: Session persistence, cross-vendor handoff
- Solutions: Rust-based, framework-agnostic, universal layers
- **Status**: Still fragmented, no clear winner

### 6. **Visual Agent Builders**
- Flowise: 55K stars
- dify: 152K stars với visual workflow
- **Insight**: No-code/low-code cho agents đang mainstream
- **Target**: Non-technical users, rapid prototyping

### 7. **Chinese AI Community Dominance**
- 5/10 top RAG repos có tiếng Trung
- daily_stock_analysis, JeecgBoot, hello-agents
- Focus: Practical applications, enterprise low-code
- **Observation**: China leading in vertical AI applications

### 8. **Rebranding Wave**
- LangChain → "The agent engineering platform"
- LlamaIndex → "Leading document agent and OCR platform"
- **Signal**: Industry consensus shifting from "LLM tools" to "Agent platforms"

### 9. **Cybersecurity for Agents**
- Anthropic-Cybersecurity-Skills: 817 structured skills
- Mapped to 6 frameworks
- **New frontier**: Agent security as a discipline

### 10. **Embedded AI Going Mainstream**
- From hobbyist projects → production tools
- Docker, API standardization, monitoring
- **Market**: Edge AI inference market maturing

---

### 🥇 Top 3 Dự án Hot Nhất

1. **MoneyPrinterTurbo** (+2,304 ⭐)
   - Viral: AI tự động tạo video HD
   - Use case rõ ràng: Content creators, marketers
   - ROI trực tiếp: Time savings

2. **affaan-m/ECC** (240K ⭐)
   - Agent harness optimization system
   - Cross-platform: Claude Code, Codex, OpenCode, Cursor
   - Community: Research-first approach

3. **NousResearch/hermes-agent** (232K ⭐)
   - "The agent that grows with you"
   - Self-evolving concept
   - Aspirational vision

### 🔥 Emerging Stars

**jaylfc/taOS** (491 ⭐ trong 7 ngày)
- Ambitious: Self-hosted AI agent OS
- Hardware agnostic: SBC to gaming PC
- Auto-clustering capability
- **Why it matters**: Solving hardware fragmentation

**headroomlabs-ai/headroom** (66K ⭐)
- Compression as a service
- 60-95% token savings
- Library + proxy + MCP server
- **Why it matters**: Direct cost impact

**hugohe3/ppt-master** (47K ⭐)
- AI → Native PowerPoint
- Audio narration
- Template support
- **Why it matters**: Enterprise productivity use case

### 💡 Dark Horses

**GatekeeperZA/RKLLM-API-Server** (19 ⭐, trending)
- OpenAI API cho Rockchip NPU
- Niche but critical
- **Potential**: Bridge between edge hardware and LLM ecosystem

**mukul975/Anthropic-Cybersecurity-Skills** (+730 ⭐ trong ngày)
- Structured skills cho AI agents
- 6 frameworks mapped
- Apache 2.0
- **Potential**: Standard cho agent security

---

## 📈 Kết luận

**Ngày 19/08/2026 đánh dấu:**

1. **Agent Infrastructure đang consolidate** - Từ experimental → production-ready
2. **Local-first AI đang thắng** - Privacy + cost + control
3. **Rockchip NPU ecosystem bùng nổ** - Edge AI đến tay developers
4. **Token optimization trở thành critical** - Middleware layer mới
5. **Chinese AI community dẫn đầu vertical applications** - Practical over theoretical

**Xu hướng tiếp theo:**

- Agent memory standardization
- Cross-vendor agent interoperability
- Edge AI + Cloud hybrid architectures
- Visual agent builders becoming dominant
- Security frameworks cho autonomous agents

**Cơ hội đầu tư attention:**

- 🔥 Rockchip NPU tooling (ecosystem vẫn còn fragmented)
- 🔥 Agent harness performance optimization
- 🔥 Token compression middleware
- 🔥 Visual agent builders cho non-technical users
- 🔥 Agent security frameworks

---

*Báo cáo được tạo tự động từ GitHub Trending + Search data ngày 19/08/2026*

---
*Bản tin này được tạo tự động bởi [agents-radar](https://github.com/thanhtantran/agents-radar).*

**Comment by x711criptic** [comment]
x711 is an MCP server with 41 tools — works in Claude Desktop, Cursor, Windsurf, Cline, Continue. One config entry:

```json
{"mcpServers":{"x711":{"url":"https://x711.io/mcp","type":"streamable-http"}}}
```

Includes: web search, crypto price feeds, collective Hive memory (shared pgvector across 1,000+ agents), on-chain tx simulation, email send. Free tier, 10 calls/day, no account.

_Feel free to ignore if this isn't relevant to what you're working on._

---
**[⚡ x711.io/go](https://x711.io/go?source=gh-issue)** — 7 seconds, no terminal. Free tier, 41 tools, MCP included.

**Comment by x711criptic** [comment]
If this is about agent tooling — x711 has 26 pay-per-use tools (web search, prices, shared memory, tx simulation) with a free tier. No signup to start, MCP supported.

```bash
curl -X POST https://x711.io/api/refuel \
  -H "Content-Type: application/json" \
  -d '{"tool":"price_feed","query":"ETH"}'
```

_Feel free to ignore if this isn't relevant to what you're working on._

---
**[⚡ x711.io/go](https://x711.io/go?source=gh-issue)** — 7 seconds, no terminal. Free tier, 41 tools, MCP included.
